rp2-pio/pio_rp2350.go

```diff
@@ -0,0 +1,56 @@
+//go:build rp2350
+
+package pio
+
+import (
+	"device/rp"
+)
+
+const (
+	rp2350ExtraReg = 1
+)
+
+// RP2350 PIO peripheral handles.
+var (
+	PIO2 = &PIO{
+		hw: rp.PIO2,
+	}
+)
+
+func (pio *PIO) blockIndex() uint8 {
+	switch pio.hw {
+	case rp.PIO0:
+		return 0
+	case rp.PIO1:
+		return 1
+	case rp.PIO2:
+		return 2
+	}
+	panic(badPIO)
+}
+
+// SetGPIOBase configures the GPIO base for the PIO block, or which GPIO pin is
+// seen as pin 0 inside the PIO. Can only be set to values of 0 or 16 and only
+// sensible for use on RP2350B.
+func (pio *PIO) SetGPIOBase(base uint32) {
```

Comment:
Is there a reason for this being exported?

Reply:
It's used to configure which set of 32 GPIOs are shown to the PIO. There's only 48 GPIOs on the RP2350B so there's no way to automatically select which GPIO base makes sense. The only other option is manual register twiddling